src/main/java/org/prebid/server/bidder/pangle/PangleBidder.java

```diff
@@ -0,0 +1,215 @@
+        final BidRequest outgoingRequest = request.toBuilder().imp(Collections.singletonList(imp)).build();
+
+        return HttpRequest.<BidRequest>builder()
+                .method(HttpMethod.POST)
+                .uri(endpointUrl)
+                .headers(makeHeaders(token))
+                .payload(outgoingRequest)
+                .body(mapper.encode(outgoingRequest))
+                .build();
+    }
+
+    private static MultiMap makeHeaders(String token) {
+        return HttpUtil.headers()
+                .add("TOKEN", token);
+    }
+
+    @Override
+    public final Result<List<BidderBid>> makeBids(HttpCall<BidRequest> httpCall, BidRequest bidRequest) {
+        try {
+            final List<BidderError> errors = new ArrayList<>();
+            final BidResponse bidResponse = mapper.decodeValue(httpCall.getResponse().getBody(), BidResponse.class);
+            return Result.of(extractBids(bidResponse, errors), errors);
+        } catch (DecodeException e) {
+            return Result.withError(BidderError.badServerResponse(e.getMessage()));
+        }
+    }
+
+    private List<BidderBid> extractBids(BidResponse bidResponse, List<BidderError> errors) {
+        if (bidResponse == null || CollectionUtils.isEmpty(bidResponse.getSeatbid())) {
+            return Collections.emptyList();
+        }
+        return bidsFromResponse(bidResponse, errors);
+    }
+
+    private List<BidderBid> bidsFromResponse(BidResponse bidResponse, List<BidderError> errors) {
+        return bidResponse.getSeatbid().stream()
+                .filter(Objects::nonNull)
+                .map(SeatBid::getBid)
+                .flatMap(Collection::stream)
+                .map(bid -> createBid(bid, bidResponse.getCur(), errors))
+                .filter(Objects::nonNull)
+                .collect(Collectors.toList());
+    }
+
+    private BidderBid createBid(Bid bid, String currency, List<BidderError> errors) {
+        final Integer adType;
+        try {
+            adType = getAdTypeFromBidExt(bid);
+        } catch (PreBidException e) {
+            errors.add(BidderError.badServerResponse(e.getMessage()));
+            return null;
+        }
+        final BidType bidType;
+        switch (adType) {
+            case 1:
+                bidType = BidType.banner;
+                break;
+            case 2:
+                bidType = BidType.banner;
+                break;
+            case 5:
+                bidType = BidType.xNative;
+                break;
+            case 7:
+                bidType = BidType.video;
+                break;
+            case 8:
+                bidType = BidType.video;
+                break;
+            default:
+                errors.add(BidderError.badServerResponse("unrecognized adtype in response"));
+                return null;
+        }
+
+        return BidderBid.of(bid, bidType, currency);
+    }
+
+    private Integer getAdTypeFromBidExt(Bid bid) {
+        if (bid == null) {
```

Comment:
Can we add `.filter(Objects::nonNull)` while looping through the bids?

Reply:
From Go implementation, they want to add errors about not present bids